mitiq/zne/scaling/folding.py

```diff
@@ -740,10 +740,19 @@ def fold_method(circuit: Circuit, scale_factor: float,**kwargs):
         this_stretch = 3.0 if scale_factor > 3.0 else scale_factor
         if fold_method_args:
             folded = fold_method(
-                folded, this_stretch, *fold_method_args, squash_moments=False
+                folded,
+                this_stretch,
+                *fold_method_args,
+                squash_moments=False,
+                **{k: v for k, v in kwargs.items() if k != "squash_moments"},
```

Comment:
What happens if instead of forcing `squash_moments` to be `False` for odd scale factors,  we set it according to `kwargs` ?
Are there bad consequences?

Reply:
I don't think there are bad consequences, but there is duplicated work. You only need to [squash moments once at the end](https://github.com/unitaryfund/mitiq/pull/525/files#diff-58c4c4e3cb99c8097c841f4267e46fe9702e78865350b7eb607653a332f77b52R758-R760).

Reply:
I see, good point!